### PR TITLE
BSP-153: Refactor BulkScan API

### PR DESCRIPTION
**Pull request by johnoneill46**
Refactor BulkScan API for FR to use the same services/models etc as Divorce. This PR also has removed any classes that have now been moved into the shared BSP Commons Lib

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
     <module name="LineLength">
         <property name="fileExtensions" value="java"/>
-        <property name="max" value="120"/>
+        <property name="max" value="150"/>
```
Comment (johnoneill46) [inline checkstyle.xml]: This is done to bring it into line with Divorce and make things easier to read

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/BulkScanValidationPatterns.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.model;
+
+public class BulkScanValidationPatterns {
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/BulkScanValidationPatterns.java]: it looks like a very good candidate to go to the lib

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/transformation/in/ExceptionRecord.java [hunk 3]
```diff
@@ -1,28 +1,29 @@
...
+import lombok.Builder;
 import lombok.Getter;
-import uk.gov.hmcts.reform.finrem.caseorchestration.model.bulkscan.OcrDataField;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.bulkscan.validation.in.OcrDataField;
 
 import java.util.List;
 
 @Getter
+@Builder
 public class ExceptionRecord {
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/transformation/in/ExceptionRecord.java]: ->lib

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/transformation/output/SuccessfulTransformationResponse.java [hunk 4]
```diff
@@ -4,23 +4,17 @@
 import lombok.Builder;
 import lombok.Data;
 
+import java.util.ArrayList;
 import java.util.List;
 
 @Data
 @Builder
 public class SuccessfulTransformationResponse {
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/transformation/output/SuccessfulTransformationResponse.java]: ->lib

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/validation/in/OcrDataField.java [hunk 5]
```diff
@@ -1,25 +1,24 @@
...
+package uk.gov.hmcts.reform.finrem.caseorchestration.model.bulkscan.validation.in;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import lombok.Data;
+import lombok.Getter;
 
-@Data
+@Getter
 public class OcrDataField {
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/validation/in/OcrDataField.java]: ->lib

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/validation/out/OcrValidationResponse.java [hunk 6]
```diff
@@ -1,25 +1,27 @@
...
+package uk.gov.hmcts.reform.finrem.caseorchestration.model.bulkscan.validation.out;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
 
 import java.util.List;
 
+@Getter
 public class OcrValidationResponse {
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/validation/out/OcrValidationResponse.java]: ->lib

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/validation/out/OcrValidationResult.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+import lombok.Getter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.commons.collections.CollectionUtils.isNotEmpty;
+
+@Getter
+public class OcrValidationResult {
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/validation/out/OcrValidationResult.java]: ->lib

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/BulkScanValidationPatterns.java (hunk 2 above)
Comment (johnoneill46) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/BulkScanValidationPatterns.java]: Moved into the BSP Commons Lib as requested

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/transformation/output/SuccessfulTransformationResponse.java (hunk 4 above)
Comment (johnoneill46) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/transformation/output/SuccessfulTransformationResponse.java]: Moved into the BSP Commons Lib as requested

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/validation/out/OcrValidationResponse.java (hunk 6 above)
Comment (johnoneill46) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/validation/out/OcrValidationResponse.java]: Moved into the BSP Commons Lib as requested

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/validation/out/OcrValidationResult.java (hunk 7 above)
Comment (johnoneill46) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/validation/out/OcrValidationResult.java]: Moved into the BSP Commons Lib as requested

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/transformation/in/ExceptionRecord.java (hunk 3 above)
Comment (johnoneill46) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/transformation/in/ExceptionRecord.java]: Moved into the BSP Commons Lib as requested

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/validation/in/OcrDataField.java (hunk 5 above)
Comment (johnoneill46) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/bulkscan/validation/in/OcrDataField.java]: Moved into the BSP Commons Lib as requested

**Pull request merged**